tutorials-v5/time-evolution/023_v5_paper-mesolve.md

```diff
@@ -0,0 +1,575 @@
+
+```python
+# Hamiltonian parameters
+Delta = 2 * np.pi  # qubit splitting
+omega_d = Delta  # drive frequency
+A = 0.01 * Delta  # drive amplitude
+
+# Bath parameters
+gamma = 0.005 * Delta / (2 * np.pi)  # dissipation strength
+temp = 0  # temperature
+
+# Simulation parameters
+psi0 = basis(2, 0)  # initial state
+e_ops = [sigmaz()]
+T = 2 * np.pi / omega_d  # period length
+tlist = np.linspace(0, 1000 * T, 500)
+```
+
+### With `mesolve`
+
+```python
+# driving field
+def f(t):
+    return np.sin(omega_d * t)
+```
+
+```python
+H0 = Delta / 2.0 * sigmaz()
+H1 = [A / 2.0 * sigmax(), f]
+H = [H0, H1]
+```
+
+```python
+c_ops_me = [np.sqrt(gamma) * sigmam()]
+```
+
+```python
+driv_res = mesolve(H, psi0, tlist, c_ops=c_ops_me, e_ops=e_ops)
+```
+
+### With Rotating-Wave-Approximated `mesolve`
+
+```python
+H_RWA = (Delta - omega_d) * 0.5 * sigmaz() + A / 4 * sigmax()
+c_ops_me_RWA = [np.sqrt(gamma) * sigmam()]
+```
+
+```python
+driv_RWA_res = mesolve(H_RWA, psi0, tlist, c_ops=c_ops_me_RWA, e_ops=e_ops)
+```
+
+### With `brmesolve`
+
+```python
+# Bose einstein distribution
+def nth(w):
+    if temp > 0:
+        return 1 / (np.exp(w / temp) - 1)
+    else:
+        return 0
+
+
+# Power spectrum
+def power_spectrum(w):
+    if w > 0:
+        return gamma * (nth(w) + 1)
+    elif w == 0:
+        return 0
+    else:
+        return gamma * nth(-w)
+```
+
+```python
+a_ops = [[sigmax(), power_spectrum]]
+```
+
+```python
+driv_br_res = brmesolve(H, psi0, tlist, a_ops, sec_cutoff=-1, e_ops=e_ops)
+```
+
+### With `HEOMSolver`
+
+```python
+wsamp = 2 * np.pi
+w0 = 5 * 2 * np.pi
+
+gamma_heom = 1.9 * w0
+
+
+lambd = np.sqrt(
+    0.5
+    * gamma
+    * ((w0**2 - wsamp**2) ** 2 + (gamma_heom**2) * ((wsamp) ** 2))
+    / (gamma_heom * wsamp)
+)
+```
+
+```python
+# TODO UnderDampedEnvironment is in a big separate on Overleaf, how to handle?
+
+# Create Environment
+# bath = UnderDampedEnvironment(lam=lambd, w0=w0, gamma=gamma_heom, T=1e-30)
+fit_times = np.linspace(0, 5, 1000)  # range for correlation function fit
+```
+
+### Comparison of Solver Results
+
+```python
+plt.figure()
+
+plt.plot(tlist, driv_res.expect[0], "-", label="mesolve (time-dep)")
+plt.plot(tlist, driv_RWA_res.expect[0], "-.", label="mesolve (rwa)")
+# plt.plot(tlist, results_corr_fit.expect[0], '--', label=r'heomsolve')
+plt.plot(tlist, driv_br_res.expect[0], ":", linewidth=3, label="brmesolve")
+
+plt.xlabel(r"$t\, /\, \Delta^{-1}$")
+plt.ylabel(r"$\langle \sigma_z \rangle$")
+plt.legend()
+plt.show()
+```
+
+### Adiabatic Energy Switching
+
+To show where usage of these local-basis collapse operators can lead to errors, we look at a single qubit system.
+This time however, the energies of the qubit are adiabatically switched between positive and negative values.
+
+$H = \dfrac{\Delta}{2} \sin{(\omega_d t)} \sigma_z$.
+
+If the drive is slow enough, we expect the bath to respond to this change and transitions from higher to lower energy levels should be induced.
```

Comment:
maybe good to add a sentence like "In this case using `mesolve` without taking into consideration the change in system energies will produce an incorrect result."